### ISSUE TITLE
Why need to sub math.log(self.noise_ratio)

### ISSUE THREAD

**Issue by gbuion** [Stonesjtu/Pytorch-NCE#12]
Hi Stonesjtu,

Thanks for the sharing this NCE implement. I have a question about details. I'd like to know why we need to sub math.log(self.noise_ratio) here:
https://github.com/Stonesjtu/Pytorch-NCE/blob/1fae107a92e24e39f25dd69b766806709c70d414/nce/nce_loss.py#L228

In this tutorial https://www.tensorflow.org/extras/candidate_sampling.pdf, see the Table of Candidate Sampling Algorithms. The input to training loss of NCE is **G(x, y) = F(x, y) - log(Q(y|x))**. 

Thanks,
Bin

**Comment by Stonesjtu** [comment]
Concretely it's derived from Bayesian posterior. Because the noise sample is *noise ratio* times  more likely to occur. You can refer to the reference paper mentioned in readme for detailed formulation.

Ideally we should be able to merge this constant term into bias of the last linear layer.
